Cython/Compiler/ExprNodes.py

```diff
@@ -5100,6 +5141,21 @@ def generate_result_code(self, code):
                     start_code,
                     stop_code,
                     code.error_goto_if_null(result, self.pos)))
+        elif self.base.type.is_array:
+            if self.to_py_func:
+                code.putln('%s = %s(%s, %s, %s, %s, %d, %d); %s' % (
+                    result,
+                    self.to_py_func,
+                    base_result,
+                    self.base.type.size,
```

Comment:
Do we make sure anywhere that we know the size of the array?

Reply:
```
elif self.base.type.is_array:
```
- means that it is carray (CArrayType) which has required parameter size. 
(init function of CArrayType)
```
 def __init__(self, base_type, size): 
 super(CArrayType, self).__init__(base_type) 
 self.size = size 
```
and value of this parameter became value of attribute self.base.type.size. So self.base.type.size must be valid.